backend/deepchecks_monitoring/logic/check_logic.py

```diff
@@ -158,6 +158,11 @@ def set_missing_frequency(cls, values: dict) -> dict:  # pylint: disable=no-self
 
         return values
 
+    def sql_all_filters(self):
+        """Create sql filter clause."""
+        # overrides TimeWindowOption.sql_all_filters
```

Comment:
`MonitorOptions` is used only in `run_check_per_window_in_range` and `run_suite_per_window_in_range`, the `TimeWindowOption.sql_all_filters` must not be applied in those functions, so it looks like a correct place where it should be overridden/

to be honest, the hierarchy of those '*Options' types is a complete overkill/ it is hard to trace what and where goes, it should be refactored, and a more compositional way should be used, but it might require a lot of work and changes as those types are used in a lot of places